source/LocalizationManager.cs

```diff
@@ -2,28 +2,36 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Globalization;
+using System.Linq;
 
 namespace VersionOne.Localization
 {
 	public class LocalizationManager
 	{
 		private readonly CultureInfo _defaultculture;
-		private readonly ITemplateSetLoader _loader;
-		private readonly string[] _setnames;
+		private readonly IEnumerable<ITemplateProvider> _providers; 
 		private readonly IDictionary _localizers;
 		private readonly IDictionary<string, string> _overrides;
 
-		public LocalizationManager(string defaultculture, ITemplateSetLoader loader, IDictionary<string, string> overrides, params string[] setnames)
-			: this(new CultureInfo(defaultculture), loader, setnames)
+		public LocalizationManager(string defaultculture, ITemplateSetLoader loader, IDictionary<string, string> overrides,
+			params string[] setnames)
+			: this(
+				new CultureInfo(defaultculture),
+				setnames.Select(setname => new CompatibilityTemplateProvider(loader, setname)))
 		{
 			_overrides = overrides;
 		}
 
 		private LocalizationManager(CultureInfo defaultculture, ITemplateSetLoader loader, params string[] setnames)
+			: this(
+			defaultculture, 
+			setnames.Select(setname => new CompatibilityTemplateProvider(loader, setname)))
+		{}
+
+		public LocalizationManager(CultureInfo defaultculture, IEnumerable<ITemplateProvider> providers)
```

Comment:
I suggest accepting just a `ITemplateProvider` argument.
The responsibility of enumerating layers of providers can be encapsulated in a `CompositeTemplateProvider`